runtime/shared/src/main/scala/org/http4s/googleapis/runtime/auth/ExternalAccountCredentials.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2024 Christopher Davenport
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s
+package googleapis.runtime.auth
+import client.Client
+import org.http4s.googleapis.runtime.auth.CredentialsFile
+import fs2.io.file.Files
+import cats.effect.Temporal
+import cats.syntax.all._
+import org.http4s.googleapis.runtime.auth.CredentialsFile.ExternalAccount.ExternalCredentialSource
+
+/** External account credentials(A.K.A Workload Identity Federation) implementation to access
+  * Google cloud resources from non-Google cloud platforms.
+  *
+  * @see
+  *   https://google.aip.dev/auth/4117
+  */
```

Comment:
ExternalAccountCredentials is the entrypoint to external account credentials authentication.
It delegates various authentications to IdentityPoolCredentials based on credential_source in a configuration file.